onset_maproom/tests/test_agronomy.py

```diff
@@ -354,9 +354,9 @@ def test_soil_plant_water_balance_with_et_crop_pd_none():
 
     assert wat_bal.p_d == pd.Timedelta(days=4)
     assert (
-        wat_bal.et.isel(T=slice(0, 4)) == wat_bal.et_crop.isel(T=slice(0, 4))
+        wat_bal.et.isel(T=slice(0, 3)) == wat_bal.et_crop.isel(T=slice(0, 3))
     ).all()
-    assert (wat_bal.et.isel(T=5) != wat_bal.et_crop.isel(T=5)).any()
+    assert (wat_bal.et.isel(T=4) != wat_bal.et_crop.isel(T=4)).any()
```

Comment:
Why did this test have to be changed? The change to how planting date is calculated was a performance improvement that wasn't supposed to affect the result.

Reply:
Yes, that is on my radar, I've done some preliminary tests but chose to push nonetheless for code review.

First, I thought I had screwed up the comparisons, as you might have noticed I am struggling often with those. That wasn't the case. Then, there is something else I need to review, maybe once we are closer to merging. There are several T grids involved in this whole thing. First, there is the T of rain, tmin, tmax (thus et) that we can assume are the same. Second, runoff is derived from a rolling, so it's T is shorter than rain's. I need to drop the heading T points since we are doing a loop, if we start with NaN, we'll just propagate NaNs. Third, there is Kc, who doesn't care about the other Ts as kc_interpolation creates it from planting date until the last day of the growing season.

Now, everything is initialized as full_like of peffective (ie taking into account the days dropped by rolling) then merged. So everybody should be on same T but Kc. You might have noticed that I select on T by location for all, but not for Kc that I select by label, using location of soil_moisture's T.

So I suspect that something was (or is still) off with those selections. I've checked the 5 tests (by printing results) before pushing and all seemed coherent. So possibly it was wrong before the push. But I'd like to give another thorough look at this when some other ongoing changes are validated.

Now I think I also need to double check the cases where the simulation is run outside the growing season as I am not confident this is covered anymore...